aspnetcore/mvc/controllers/routing.md

```diff
@@ -70,667 +65,788 @@ public class HomeController : Controller
+ASP.NET Core has the following HTTP verb templates:
 
-```csharp
-[HttpGet("/products")]
-public IActionResult ListProducts()
-{
-   // ...
-}
+* [[HttpGet]](xref:Microsoft.AspNetCore.Mvc.HttpGetAttribute)
+* [[HttpPost]](xref:Microsoft.AspNetCore.Mvc.HttpPostAttribute)
+* [[HttpPut]](xref:Microsoft.AspNetCore.Mvc.HttpPutAttribute)
+* [[HttpDelete]](xref:Microsoft.AspNetCore.Mvc.HttpDeleteAttribute)
+* [[HttpHead]](xref:Microsoft.AspNetCore.Mvc.HttpHeadAttribute)
+* [[HttpPatch]](xref:Microsoft.AspNetCore.Mvc.HttpPatchAttribute)
 
-[HttpPost("/products")]
-public IActionResult CreateProduct(...)
-{
-   // ...
-}
-```
+<a name="rt"></a>
 
-For a URL path like `/products` the `ProductsApi.ListProducts` action will be executed when the HTTP verb is `GET` and `ProductsApi.CreateProduct` will be executed when the HTTP verb is `POST`. 
+### Route templates
 
-> [!TIP]
-> When building a REST API, it's rare that you will want to use `[Route(...)]` on an action method as the action will accept all HTTP methods. It's better to use the more specific `Http*Verb*Attributes` to be precise about what your API supports. Clients of REST APIs are expected to know what paths and HTTP verbs map to specific logical operations.
-> REST APIs should use attribute routing to model the app's functionality as a set of resources where operations are represented by HTTP verbs. This means that many operations (for example, GET, POST) on the same logical resource will use the same URL. Attribute routing provides a level of control that's needed to carefully design an API's public endpoint layout.
+ASP.NET Core has the following route templates:
 
-Since an attribute route applies to a specific action, it's easy to make parameters required as part of the route template definition. In this example, `id` is required as part of the URL path.
+* All the [HTTP verb templates](#verb) are route templates.
+* [[Route]](xref:Microsoft.AspNetCore.Mvc.RouteAttribute)
 
-```csharp
-public class ProductsApiController : Controller
-{
-   [HttpGet("/products/{id}", Name = "Products_List")]
-   public IActionResult GetProduct(int id) { ... }
-}
-```
+<a name="arx"></a>
 
-The `ProductsApi.GetProduct(int)` action will be executed for a URL path like `/products/3` but not for a URL path like `/products`. See [Routing](xref:fundamentals/routing) for a full description of route templates and related options.
+### Attribute routing with Http verb attributes
 
-## Route Name
+Unlike routes created for controllers with views that generally include `[action]` in the route template, API controllers typically don't include `[action]` in the route template. Because `[action]` isn't in the route template, the [action](#action) name is not in the route. That is, the method name isn't used in the matching route.
```

Comment:
I'm not super sure what this is trying to explain and why it's so important. We literally have a feature that does this, and we just showed its use. 

I don't think we need to discourage people from doing this, I can think of cases where I would use it even in a RESTful API.